### PR TITLE
Fix Android build, refactor build pipeline, add Android build to PR validation

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

Splits out the builds into different jobs, adds an Android build to PR validation, and fix the Android build.

### CODE REVIEW

**Review by MaxWang-MS — DISMISSED** [review]

File: UnityProjects/MRTKDevTemplate/Assets/XR/Settings/OpenXR Package Settings.asset [hunk 1]
```diff
@@ -410,6 +411,36 @@ MonoBehaviour:
   company: Unity
   priority: 0
   required: 0
+--- !u!114 &-243070923169931762
```
Comment (Zee2) [inline UnityProjects/MRTKDevTemplate/Assets/XR/Settings/OpenXR Package Settings.asset:414]: Based on our discussion today, should we hold off on applying this change, as the MR OpenXR Plugin does not work with the new Meta Quest feature name (broken manifest, etc)? @keveleigh

**Review by Zee2 — CHANGES_REQUESTED** [review]
Blocking on broken manifest issue as we discussed IRL today @keveleigh @Clandguth

File: UnityProjects/MRTKDevTemplate/Assets/XR/Settings/OpenXR Package Settings.asset (hunk 1 above)
Comment (keveleigh) [reply UnityProjects/MRTKDevTemplate/Assets/XR/Settings/OpenXR Package Settings.asset:414]: Works for me! That update is available in #11303, so I'll mark this as DO NOT MERGE until that PR goes in.

Comment (keveleigh) [reply UnityProjects/MRTKDevTemplate/Assets/XR/Settings/OpenXR Package Settings.asset:414]: @Zee2 This one should be ready now!

**Review by davidkline-ms — APPROVED** [review]

**Comment by Zee2** [comment]
Unblocking as per 11303 upgrading to MR OpenXR Plugin 1.7.0 fixing manifest issues.

**Review by Zee2 — APPROVED** [review]

**Pull request merged**